### PR TITLE
Cleardata for RamDisk and Volatile, and stop all arcs before clearing…

### PR DESCRIPTION
**Pull request by galganif**
… the data.

### CODE REVIEW

File: javatests/arcs/android/storage/service/StorageServiceManagerTest.kt [hunk 1]
```diff
@@ -47,49 +46,67 @@ import kotlin.coroutines.coroutineContext
...
         SchemaRegistry.register(DummyEntity)
     }
 
     @Test
     fun clearAll() = runBlocking {
-        val handle = createSingletonHandle()
+        testClearAllForKey(databaseKey)
+        testClearAllForKey(ramdiskKey)
+        testClearAllForKey(volatileKey)
+    }
```
Comment (jasonwyatt) [inline javatests/arcs/android/storage/service/StorageServiceManagerTest.kt:77]: These should ideally be separate tests.

File: javatests/arcs/android/storage/service/StorageServiceManagerTest.kt [hunk 2]
```diff
@@ -47,49 +46,67 @@ import kotlin.coroutines.coroutineContext
...
+        // Create a new handle (with new Entity manager) to confirm data is gone from storage.
+        assertThat(createSingletonHandle(storageKey).fetch()).isNull()
     }
 
     @Test
-    fun clearDataBetween() = runBlocking<Unit> {
+    fun clearDataBetween() = runBlocking {
+        testClearDataBetweenForKey(databaseKey, allRemoved = false)
+        testClearDataBetweenForKey(ramdiskKey, allRemoved = true)
+        testClearDataBetweenForKey(volatileKey, allRemoved = true)
```
Comment (jasonwyatt) [inline javatests/arcs/android/storage/service/StorageServiceManagerTest.kt:101]: same

File: javatests/arcs/android/storage/service/StorageServiceManagerTest.kt [hunk 3]
```diff
@@ -100,43 +117,45 @@ class StorageServiceManagerTest {
...
-        withContext(handle.dispatcher) {
-            assertThat(handle.fetchAll()).containsExactly(entity1, entity3)
+
+        // Create a new handle (with new Entity manager) to confirm data is gone from storage.
+        if(allRemoved) {
+            assertThat(createCollectionHandle(storageKey).fetchAll()).isEmpty()
+        } else {
+            assertThat(createCollectionHandle(storageKey).fetchAll())
+                .containsExactly(entity1, entity3)
         }
```
Comment (jasonwyatt) [inline javatests/arcs/android/storage/service/StorageServiceManagerTest.kt:130]: Consider creating the handle, then wrapping the assertions within `withContext(handle.dispatcher)`.

File: java/arcs/android/storage/service/StorageServiceManager.kt [hunk 4]
```diff
@@ -30,8 +30,10 @@ class StorageServiceManager(
     private val coroutineContext = parentCoroutineContext + CoroutineName("StorageServiceManager")
 
     override fun clearAll(resultCallback: IResultCallback) {
-        CoroutineScope(coroutineContext).launch {
+        runBlocking(coroutineContext) {
+            ArcHostManager.pauseAllHosts()
             DriverFactory.removeAllEntities().join()
+            ArcHostManager.unPauseAllHosts()
```
[the remark this answers is not in the document]
Comment (galganif) [reply java/arcs/android/storage/service/StorageServiceManager.kt]: Nice! I changed ArcHostManager this way, looks much better. Let me know if you want me to split the ArcHostManager changes to another pr.

The other comments are all addressed as well, ptal.

**Comment by galganif** [comment]
> Looks great, the only concern I have is whether or not pauseAllHostsFor will work cross-process?

As discussed it should work, as intents are sent to Android arc hosts. I will try to setup an integration test for everything as well.

**Pull request merged**